RadixWallet/Core/FeaturePrelude/AddressView/AddressDetails.swift

```diff
@@ -0,0 +1,187 @@
+// MARK: - AddressDetails
+public struct AddressDetails: Sendable, FeatureReducer {
+	public struct State: Sendable, Hashable {
+		let address: LedgerIdentifiable.Address
+		let closeAction: (() -> Void)?
```

Comment:
It really bothers me that we need to send a non-sendable closure to this state.. But it is the only way I was able to handle dismissal when the view is presented from `AddressView` (which is a plain SwiftUI vanilla view). Open to better suggestions.

Reply:
Doesn't `@Environment(\.dismiss) var dismiss` work (in the View)? Or `@Dependency(\.dismiss) var dismiss`?

Reply:
I mean the reducer doesn't need to know that we closed it, right? It could just be done on the View side?

Reply:
I literally don't understand what was the problem back then what I added this.. removing the whole closure concept.